pkg/apis/core/v1/zz_generated.validations.go

```diff
@@ -284,6 +288,23 @@ func Validate_EndpointsList(in *v1.EndpointsList, fldPath *field.Path) (errs fie
 	return errs
 }
 
+func Validate_EnvVar(in *v1.EnvVar, fldPath *field.Path) (errs field.ErrorList) {
+	if in.ValueFrom != nil {
+		errs = append(errs, Validate_EnvVarSource(in.ValueFrom, fldPath.Child("valueFrom"))...)
+	}
+	return errs
+}
+
+func Validate_EnvVarSource(in *v1.EnvVarSource, fldPath *field.Path) (errs field.ErrorList) {
+	errs = append(errs, validate.Union(fldPath, in, map[any]any{
+		"resourceFieldRef": []any{in.ResourceFieldRef},
+		"configMapKeyRef":  []any{in.ConfigMapKeyRef},
+		"secretKeyRef":     []any{in.SecretKeyRef},
+		"fieldRef":         []any{in.FieldRef},
+	})...)
+	return errs
```

Comment:
There's probably a better way to do this.  What I was looking for was a simple way to provide:

The values of the fields in a struct that were tagged with `+unionMember`, grouped by the unionMember those fields belong to (multiple fields can belong to the same union member, this all aligns with the Unions KEP).

Reply:
one possible alternative is to generate all of the union enforcing code rather than building a structure to pass to a validator function

Reply:
I'm currently favoring:

```go
var envVarSourceUnionMembership = validate.NewUnionMembership("fieldRef", "resourceFieldRef", "configMapKeyRef", "secretKeyRef")

func Validate_EnvVarSource(in *v1.EnvVarSource, fldPath *field.Path) (errs field.ErrorList) {
	errs = append(errs, validate.Union(fldPath, in, envVarSourceUnionMembership, in.FieldRef, in.ResourceFieldRef, in.ConfigMapKeyRef, in.SecretKeyRef)...)
	return errs
}
```

This complicates codegen significantly, but it's general purpose, and enums could also use it to generate a var holding the set of supported values.